src/backend/distributed/planner/query_pushdown_planning.c

```diff
@@ -1539,18 +1546,12 @@ SubqueryPushdownMultiNodeTree(Query *originalQuery)
 	 * columnList. Columns mentioned in multiProject node and multiExtendedOp
 	 * node are indexed with their respective position in columnList.
 	 */
-	List *targetColumnList = pull_var_clause_default((Node *) targetEntryList);
+	List *targetColumnList = pull_vars_of_level((Node *) targetEntryList, 0);
```

Comment:
I must be missing something on a Friday evening, but I couldn't understand how `levelsup = 0` works here:

```C
	elog(INFO, "pull_vars_of_level: %s", nodeToString(pull_vars_of_level((Node *) targetEntryList, 0)));
	elog(INFO, "pull_var_clause_default: %s", nodeToString(pull_var_clause_default(targetEntryList)));
```
And,
```SQL
SELECT event_type, (SELECT 1 + 3 + e.value_2)
FROM events_table e
ORDER BY 1,2 LIMIT 1;
INFO:  pull_vars_of_level: ({VAR :varno 1 :varattno 3 :vartype 23 :vartypmod -1 :varcollid 0 :varlevelsup 0 :varnoold 1 :varoattno 3 :location 7} {VAR :varno 1 :varattno 4 :vartype 23 :vartypmod -1 :varcollid 0 :varlevelsup 1 :varnoold 1 :varoattno 4 :location 35})
INFO:  pull_var_clause_default: ({VAR :varno 1 :varattno 3 :vartype 23 :vartypmod -1 :varcollid 0 :varlevelsup 0 :varnoold 1 :varoattno 3 :location 7})

```

`pull_vars_of_level ` call pulls a var with `varlevelsup 1`. How could that work?

Reply:
The name is a bit misleading. It gets all vars that point to level 0, so as it descends into a subquery it raises the level it's looking for by 1.